### ISSUE TITLE
Error when trying to compare read-only files.

### ISSUE THREAD

**Issue by GregSherman** [moshfeu/vscode-compare-folders#128]
![Screenshot 2023-08-18 at 12 41 53 PM](https://github.com/moshfeu/vscode-compare-folders/assets/63402349/12777bb4-3d84-40c6-99f7-03480fa6e1ee)

Upon trying to compare folders with read-only files, I get this message. Not sure if it's something someone can get around, but since a diff tool really only reads files, I feel like it should be able to compare them.

**Comment by gliviu** [comment]
Happens on windows or linux?

**Comment by GregSherman** [comment]
macOS, files are 0o444 perms

**Comment by gliviu** [comment]
Most probably it is a permission denied error while comparing the folders. I tried the 0444 on linux and I got EACCESS errors. 

[0444 does not allow lstat on files inside read-only directories](https://unix.stackexchange.com/questions/584484/set-folder-permissions-to-0444-but-now-cannot-cd-in). Lstat is required by the folder compare algorithm to determine if an entry is a file or directory.

To see if this is your problem, take a look at the actual error. This is how to do it:
![image](https://github.com/moshfeu/vscode-compare-folders/assets/8597812/36b24720-ae5a-451d-aa5c-25b07076d648)
